clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEvent.java

```diff
@@ -21,18 +21,27 @@
  */
 abstract public class ApplicationEvent {
     public final Type type;
+    protected final String owner;
 
-    protected ApplicationEvent(Type type) {
+    protected ApplicationEvent(Type type, String owner) {
```

Comment:
As a simplification, this is not needed I would say. We could just remove the owner var and constructor param, and leave only a simple `toStringBase` like this:
```
protected String toStringBase() {
        return "owner='" + getClass().getSimpleName() + '\'' +
            ", type=" + type;
 }
```
It would end up returning the simple name of the child class, which is what we want in the end.

Reply:
maybe we could get rid of owner after all, and just have the calling class to print its name so that we could follow a more conventional toString().  Here's an example of how it might look like after getting rid of owner.  Would this work?
```
ListOffsetsApplicationEvent{type=LIST_OFFSETS, future=java.util.concurrent.CompletableFuture@bdd2027[Not completed], timestampsToSearch={}, requireTimestamps=false}
```

Reply:
Sure, the point was to avoid the duplicated info in the toString, result of including the class name in the base class and in the child implementations. See my previous [comment](https://github.com/apache/kafka/pull/14391#discussion_r1327561641) related to this down below.